main.go

```diff
@@ -74,8 +74,7 @@ func main() {
 		os.Exit(0)
 	}
 
-	// TODO: Probably not necessary to pass all of opts into builder, right?
-	collectorBuilder := kcollectors.NewBuilder(context.TODO(), opts)
+	collectorBuilder := kcollectors.NewBuilder(context.TODO())
```

Comment:
this should probably be a cancellable background context that is cancelled on shutdown, I'd say we take care of that in a follow up, but can you open an issue to track this?